robot_smach_states/src/robot_smach_states/human_interaction/find_people_in_room.py

```diff
@@ -0,0 +1,532 @@
+        property has non None values.
+            If set to True then only people with all specified
+            properties are returned, else all people with at least one true property.
+            Example:
+                properties = {'tags': ['LWave', 'RWave', 'LHolding', 'RHolding']}
+                strict = True
+                    This will return a list of people who have the tags:
+                        'LWave' AND 'RWave' AND 'LHolding' AND 'RHolding'
+
+                strict = False
+                    This will return a list of people who have the tags:
+                        'LWave' OR 'RWave' OR 'LHolding' OR 'RHolding'
+
+        :param nearest: (bool) (default: False) If True, selects the people nearest to the robot who match the
+            requirements posed using the properties, query_entity_designator, look distance and strict arguments
+        :param attempts: (int) (default: 1) Max number of search attempts
+        :param search_timeout: (float) (default: 60) maximum time the robot is allowed to search
+        :param look_range: (tuple of size 2) (default: (-np.pi/2, np.pi/2)) from what to what head angle should the
+        robot search
+        :param look_steps: (int) (default: 8) How many steps does it take in that range
+        """
+        super(SetPoseFirstFoundPersonToEntity,
+              self).__init__(outcomes=["done", "failed"])
+
+        if not found_person_designator:
+            found_person_designator = ds.VariableDesignator(resolve_type=Entity, name='new_person').writeable
+
+        with self:
+            self.add("FIND_FIRST_PERSON",
+                     FindFirstPerson(
+                         robot=robot,
+                         found_person_designator=found_person_designator,
+                         properties=properties,
+                         query_entity_designator=query_entity_designator,
+                         look_distance=look_distance,
+                         speak=speak,
+                         strict=strict,
+                         nearest=nearest,
+                         attempts=attempts,
+                         search_timeout=search_timeout,
+                         look_range=look_range,
+                         look_steps=look_steps),
+                     transitions={
+                         'found': 'UPDATE_POSE',
+                         'failed': 'failed'
+                     })
+
+            self.add("UPDATE_POSE",
+                     states.UpdateDestEntityPoseWithSrcEntity(
+                         robot=robot,
+                         src_entity_designator=found_person_designator,
+                         dst_entity_designator=dst_entity_designator,
+                         dst_entity_type=dst_entity_type),
+                     transitions={
+                         'done': 'done',
+                         'failed': 'failed'
+                     })
+
+
+class _DecideNavigateState(smach.State):
```

Comment:
I dont think there should be a separate state for deciding this. This could be resolved in the initialisation of the statemachine.